### PR TITLE
Remove some unused crate dependencies.

### PR DESCRIPTION
**Pull request by nnethercote**
I found these by setting the `unused_crate_dependencies` lint temporarily to `Warn`.

r? @jackh726

### CODE REVIEW

**Comment by nnethercote** [comment]
There's an argument to be made that we should add `#[deny(unused_crate_dependencies)]` to most of the crates in the compiler, but I haven't done that here.

File: library/sysroot/src/lib.rs [hunk 1]
```diff
@@ -1 +1,4 @@
 // This is intentionally empty since this crate is only used to depend on other library crates.
+
```
Comment (jackh726) [inline library/sysroot/src/lib.rs:2]: Why this change?

**Review by jackh726 — COMMENTED** [review]
Can you make an MCP to change the lint to deny?

**Comment by mati865** [comment]
I think it could be enabled in in `[lints]` section of Cargo.toml. This should be more manageable than adding `deny` to each lib.rs.

**Comment by nnethercote** [comment]
Something I didn't say is that when I enabled it universally I had to deal with several annoying cases, mostly in `library/`.
- Some crates are only used in certain configurations.
- Some crates are in `Cargo.toml` because of weird packaging reasons. E.g. https://github.com/rust-lang/rust/blob/e1ac0fa95ba15221ee195647cb12464df3164513/compiler/rustc/Cargo.toml#L9-L20

These need `use rustc_foo as _;` items to avoid the warning, or some fiddling with conditional compilation stuff. Both of the cases you asked about above were leftovers for that kind of thing that I failed to removed.

So my conclusion is that this would be good to enable for almost every compiler crate (except the `rustc` crate), but not `library/`. There are also these false positive cases: https://github.com/rust-lang/rust/blob/e1ac0fa95ba15221ee195647cb12464df3164513/compiler/rustc_lint_defs/src/builtin.rs#L533-L539

So I think changing it to `Deny` in general is not a good idea.

**Comment by nnethercote** [comment]
> I think it could be enabled in in `[lints]` section of Cargo.toml. This should be more manageable than adding `deny` to each lib.rs.

I tried adding this to the top level `Cargo.toml`:
```toml
[workspace.lints.rust]
unused_crate_dependencies = "warn"
```
and it had no effect, even on a stage 2 build. I also tried without the `workspace.` but then I got a Cargo error. Any other suggestions? I'm no expert on Cargo and workspaces.

**Comment by mati865** [comment]
When you add it to the workspace, you also have to enable lints inheritance in each member:
```
[lints]
workspace = true
```
https://doc.rust-lang.org/cargo/reference/workspaces.html#the-lints-table

**Comment by nnethercote** [comment]
I rebased.

**Comment by jackh726** [comment]
@bors r+

As much as I am wary about landing one-off changes like this without also *linting* to prevent similar PRs in the future, sounds like there are real reasons we can't lint (at the moment at least). So, I'm fine to land this.

**Comment by jackh726** [comment]
(I would still recommend following up with linting for compiler crates if possible)

**Comment by nnethercote** [comment]
> When you add it to the workspace, you also have to enable lints inheritance in each member:

Would I have to make that change to the `Cargo.toml` file of every `rustc_*` crate under `compiler/`? If so, that's worse than adding a single `#![deny(unused_crate_dependencies)]` line to every `lib.rs` file.
